cmd/internal/bpm_configs.go

```diff
@@ -25,71 +27,103 @@ This command calculates and prints the BPM configurations for all all BOSH jobs
 instance group.
 `,
 	RunE: func(cmd *cobra.Command, args []string) error {
-		log = newLogger()
-		defer log.Sync()
-		boshManifestPath := viper.GetString("bosh-manifest-path")
-		if len(boshManifestPath) == 0 {
-			return fmt.Errorf("manifest cannot be empty")
-		}
-
-		baseDir := viper.GetString("base-dir")
-		if len(baseDir) == 0 {
-			return fmt.Errorf("base directory cannot be empty")
-		}
-
-		namespace := viper.GetString("cf-operator-namespace")
-		if len(namespace) == 0 {
-			return fmt.Errorf("namespace cannot be empty")
-		}
-
-		instanceGroupName := viper.GetString("instance-group-name")
-		if len(instanceGroupName) == 0 {
-			return fmt.Errorf("instance-group-name cannot be empty")
-		}
-
-		boshManifestBytes, err := ioutil.ReadFile(boshManifestPath)
-		if err != nil {
-			return err
-		}
-
-		m, err := manifest.LoadYAML(boshManifestBytes)
-		if err != nil {
-			return err
-		}
-
-		dg, err := manifest.NewDataGatherer(log, baseDir, namespace, *m, instanceGroupName)
-		if err != nil {
-			return err
-		}
-
-		bpmConfigs, err := dg.BPMConfigs()
-		if err != nil {
-			return err
-		}
-
-		bpmBytes, err := yaml.Marshal(bpmConfigs)
-		if err != nil {
-			return errors.Wrapf(err, "could not marshal json output")
-		}
-
-		jsonBytes, err := json.Marshal(map[string]string{
-			"bpm.yaml": string(bpmBytes),
-		})
-		if err != nil {
-			return errors.Wrapf(err, "could not marshal json output")
-		}
-
-		f := bufio.NewWriter(os.Stdout)
-		defer f.Flush()
-		_, err = f.Write(jsonBytes)
-		if err != nil {
-			return err
-		}
-
-		return nil
+		return gatherInstanceGroupBPMConfigs()
```

Comment:
I'm not a huge fan of this private function. We're not going to call it from somewhere else, it doesn't help documenting the code. Also it seems to break the 'cobra' style, which we're using in all the other commands.

We're already passing a function into the struct: `RunE: func(cmd *cobra.Command, args []string) error {`

Reply:
I will revert.